tests/plugins/widget/widgetsintegration.js

```diff
@@ -673,6 +664,8 @@
 					assert.isTrue( !!getWidgetById( editor, 'w1' ), 'widget has not been deleted' );
 					assert.isFalse( !!editor.getSelection().isFake, 'selection is not faked' );
 					assert.isFalse( !!editor.document.getById( 'cke_copybin' ), 'copybin was removed' );
+
+					this.editor.setReadOnly( false );
```

Comment:
If some of the assertions above failed, then editor will remain read-only after this test. It's safer to keep it in `tearDown` method.